analyses/cell-type-ewings/04-infercnv.Rmd

```diff
@@ -0,0 +1,433 @@
+chr_columns_to_keep <- which(!stringr::str_detect(names(coldata_df), "chrMT|chrGL000219.1"))
+cnv_df <- coldata_df[, chr_columns_to_keep]
+```
+
+The metadata that we added includes the following columns for each chromosome: 
+
+- `has_cnv_chr1`
+- `has_loss_chr1`
+- `has_dupli_chr1`
+- `proportion_cnv_chr1`
+- `proportion_loss_chr1`
+- `proportion_dupli_chr1`
+- `proportion_scaled_cnv_chr1`
+- `proportion_scaled_loss_chr1`
+- `proportion_scaled_dupli_chr1`
+
+The `has` columns are boolean columns indicating if that cell has presence of a CNV, loss, or duplication event. 
+If `has_cnv` is TRUE if either `has_loss` or `has_dupli` is TRUE. 
+
+See [the comments in this issue](https://github.com/broadinstitute/infercnv/issues/208) explaining what the contents of the columns are. 
+
+One approach to classifying tumor cells is to look for cells that have a lot of CNVs. 
+Let's start by just summing up the total CNVs/cell and plotting that. 
+
+
+```{r}
+# grab any columns that have has_cnv 
+has_cnv_cols <- names(cnv_df)[startsWith(colnames(cnv_df), "has_cnv")]
+
+cnv_df <- cnv_df |>
+  # get total number of cnvs per cell and then scale that number 
+  dplyr::mutate(has_cnv = rowSums(across(has_cnv_cols)),
+                scaled_cnv = scale(has_cnv))
+```
+
+
+```{r}
+# plot the total number of CNVs
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = has_cnv)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  scale_color_viridis_c()
+```
+This shows that the total CNVs are highest in the group of cells that correspond to the same cells that are manually annotated as tumor cells, which is promising. 
+If we were to use total CNVs we would need some sort of cut off for the number. 
+Let's look at the distribution of total CNVs and also the scaled total CNVs to see if we can identify a cut off. 
+
+```{r}
+ggplot(cnv_df, aes(x = has_cnv)) +
+  geom_density() +
+  theme_bw() +
+  labs(x = "Total number of CNVs")
+
+ggplot(cnv_df, aes(x= scaled_cnv)) + 
+  geom_density() +
+  theme_bw() +
+  labs(x = "Scaled number of CNVS")
+```
+
+Looking at the scaled number of CNVs it does look like we could use 0 as the cutoff and call any cell that has > 0 CNVs (after scaling) as tumor and all other cells as normal. 
+
+```{r}
+# add cnv classification 
+cnv_df <- cnv_df |> 
+  dplyr::mutate(cnv_classification = dplyr::if_else(scaled_cnv > 0, "Tumor", "Normal"))
+
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = cnv_classification)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() 
+```
+
+Another option is to use the scaled proportion of CNVs which tells us the proportion of genes on that chromosome that are part of the CNV.  
+
+```{r}
+proportion_cols <- names(cnv_df)[startsWith(colnames(cnv_df), "proportion_scaled_cnv")]
+
+# get mean proportion of cnvs per cell and then scale that number 
+cnv_df <- cnv_df |> 
+  dplyr::mutate(mean_proportion = rowMeans(across(proportion_cols)),
+                scaled_mean_proportion = scale(mean_proportion))
```

Comment:
Since proportions are already bounded 0-1, I probably would not use scaling here. It probably doesn't quite apply to counting chromosomes either, now that I think about it.